packages/core/test/buttons/buttonTests.tsx

```diff
@@ -84,5 +95,42 @@ function buttonTestSuite(component: React.ComponentClass<any>, tagName: string)
             const element = React.createElement(component, props);
             return useMount ? mount(element) : shallow(element);
         }
+
+        function checkClickTriggeredOnKeyUp(done: MochaDone,
+                                            buttonProps: Partial<IButtonProps>,
+                                            keyEventProps: Partial<React.KeyboardEvent<any>>) {
+            const wrapper = button(buttonProps, true);
+
+            // private members in TS are still accessible, and since Enzyme
+            // doesn't properly propagate events on non-string refs
+            // (https://github.com/airbnb/enzyme/issues/566), we need to dig in
+            // and attach a spy to the ref's .click function ourselves.
+            const buttonRef = (wrapper.instance() as any).buttonRef;
+            const onClick = sinon.spy(buttonRef, "click");
+
+            wrapper.simulate("keyup", keyEventProps);
+
+            // wait for the whole lifecycle to run
+            setTimeout(() => {
+                assert.equal(onClick.callCount, 1);
+                done();
```

Comment:
im pretty sure you do, but did you check if this is still needed with the change in the logic?